### PR TITLE
Add parameter to allow case insensitive matches

### PR DESCRIPTION
**Pull request by vladbarosan**
Add the option to to do case insensitive regex matches.

### CODE REVIEW

**Comment by vladbarosan** [comment]
@amarzavery  this is required to enable this in the live validator. A lot of traffic is getting operation not found because they don't respect the cases but the services themselves don't seem to have a problem with this as they accept the request.

File: lib/types/api.js [hunk 1]
```diff
@@ -60,7 +60,7 @@ var validators = require('../validation/validators');
  *
  * @constructor
  */
-function SwaggerApi (definition, definitionRemotesResolved, definitionFullyResolved, references, options) {
```
Comment (amarzavery) [inline lib/types/api.js]: Can you make sure to not have the unnecessary format changes. This causes merge conflicts for us when we want to merge changes from upstream master.

Comment (vladbarosan) [reply lib/types/api.js]: @amarzavery  yea Ill try, will have to do it in notepad or something, otherwise i need to desactivate everything in VS code

Comment (vladbarosan) [reply lib/types/api.js]: Ill push an update with removing the formatting

Comment (amarzavery) [reply lib/types/api.js]: you can remove auto formatting on save from your settings.json file in .vscode temporarily for this change and then enable it back.

Comment (vladbarosan) [reply lib/types/api.js]: I tried but didnt work, i think I need to restart all VS instances and I have like 6 opened :(

**Comment by vladbarosan** [comment]
@amarzavery  done

**Pull request merged**